### PR TITLE
Adding CICD pipeline dashboard

### PR DESCRIPTION
**Pull request by bblinder**
A dashboard to monitor a CI/CD pipeline deployment. Many variables like chart names and signal names are there for demonstration purposes and can be adjusted to fit the specific use case.

### CODE REVIEW

**Comment by bblinder** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by bblinder** [comment]
I have read the Code of Conduct and I hereby accept the Terms

**Comment by rcastley** [comment]
(Copied from closed PR) - The generator behind the CI/CD pipeline demo is not public, this is a public repo so just wondering how the public would use/benefit from this.

Internally, this is awesome just trying to understand the benefits of inclusion in a public repo.

**Comment by bblinder** [comment]
> (Copied from closed PR) - The generator behind the CI/CD pipeline demo is not public, this is a public repo so just wondering how the public would use/benefit from this.
> 
> Internally, this is awesome just trying to understand the benefits of inclusion in a public repo.

I'm not too worried about the data generation behind the scenes for this demo, or whether it works outside its demo environment. The idea is to show an example of a custom dashboard that speaks to a particularly resonant use-case. 

_"We didn't know you could even do that"_ is a common refrain when shown to prospects. The idea that you could show -- in real time -- metrics, logs, and notable events around a code push, and most importantly having business context around all of these, is a major differentiator that has sparked some good ideas around what's possible.

Caveating that this is an example of an "optimized" workflow, I'm happy to make it even more general/templatized so it can be made to work more easily in any environment, provided the end user has an idea of what they want to monitor. To get it to "minimally viable" and somewhat working OOTB, we can tweak it to include default dimensions or signals that would likely already be in an environment.

Thoughts?

File: dashboards-and-dashboard-groups/CICD-Pipeline/variables.tf [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+variable "log_observer_connect_index" {
+  description = "The Splunk Enterprise/Cloud index. Can be more than one."
+  type        = string
+}
+
+variable "push_event_name" {
+  description = "The event name to push to the event feed"
+  default     = "canary push event"
+  type        = string
+}
```
Comment (greatestusername-splunk) [inline dashboards-and-dashboard-groups/CICD-Pipeline/variables.tf:55]: Might be worth adding a section to the `README.md` calling out "Events" and this variable and how the dashboard uses events? 
Users might like that context about why they would use events and which events would be useful for this dashboard.

File: dashboards-and-dashboard-groups/CICD-Pipeline/CICD_pipeline.tf [hunk 2]
```diff
@@ -0,0 +1,476 @@
...
+  auth_token = var.access_token
+}
+
+# signalfx_event_feed_chart.CICD_pipeline_0:
+resource "signalfx_event_feed_chart" "CICD_pipeline_0" {
+  name         = "Event Feed"
+  program_text = <<-EOF
+        C = events(eventType='Automated Rollback initiated').publish(label='C')
+        B = events(eventType='${var.push_event_name}').publish(label='B')
+        A = alerts(detector_name='CI/CD Service Outlier').publish(label='A')
```
Comment (greatestusername-splunk) [inline dashboards-and-dashboard-groups/CICD-Pipeline/CICD_pipeline.tf:12]: Might be worth calling out these static named events/alerts in the `README.md` and how they are useful. 

Might also consider making them changable as a variable so users can add their own "automated roll back" events and the detector name they want?

**Review by greatestusername-splunk — CHANGES_REQUESTED** [review]
Couple asks re: documentation and variables. Specifically how are Events used/useful and can they be described more in the docs?

**Pull request closed without merging**